### ISSUE TITLE
Equilibrage du jeu - espèces (bio/meca), production, maintenance et saturation

### ISSUE THREAD

**Issue by LoucIsZbib** [LoucIsZbib/SpaceBaracraftCivilization#1]
Pour SBC, j'aurai souhaité avoir plus de variété dans les espèces que ce que propose Farhorizons (où globalement les mêmes planètes sont intéressantes pour tout le monde).
J'ai imaginé un truc avec 2 natures différentes : soit bio, soit méca (soit un mélange des deux). Forcément, les biomes ne seront pas les mêmes : le chaud humide est favorable aux êtres vivants (par exemple avec plus de pousse des végétaux), alors que c'est le cauchemar pour des machines, à cause de la rouille.
En plus, ça permettrait d'avoir des zergs sur ce jeu :)

Concrètement, il y aura pas un LSN par planète, mais un "factor" orienté food, et un factor orienté méca, qui rendraient une planète plus adaptée à une espèce bio ou méca.
Le niveau de tech en bio ou méca permettrait de rendre l'espece plus adaptable à d'autres climats.

Et pour ouvrir des différentes voies de dev, je me suis dit que le biologique avait toujours le meme cout d'entretien (combien  tu manges ne dépend pas de l'environnement), mais une productivité qui dépend du climat (imaginez les champs de blés au sahara). Et le méca aurait une prod constante (la cadence des robots), mais un cout d'entretien dépendant du climat (rouille).
On peut imaginer des augmentations de productivité en fonction des technos, et/ou des baisses de cout d'entretien en fonction des progres technos. J'ai pas encore intégré ça dans la balance du jeu. A voir.

Je n'ai pas non plus imaginé ce que peut donner une espèce hybride, comme les humains : créatures bio qui utilisent des technos méca. A réfléchir

J'ai ajouté une saturation en fonction de la population (bio ou méca). L'idée est d'avoir une progression sur toutes les colonies (comme avec des CU / IU), y compris la home, mais brider pour ne pas avoir de croissance exponentielle infinie, notamment pour forcer à aller coloniser ailleurs, et permettre de garder les joueurs dans une fourchette de puissance rapprochée.
En pratique, on définit une pop cible (dans ma simu 2000) qui pourrait être un critère des planètes, par exemple la taille, la richesse naturelle. Et on calcule une exp(-pop/cible), si la pop vaut 0, la saturation vaut 1 (= productivité max), si pop tend vers l'infini, la saturation vaut 0 (et la productivité vaut 0). En termes de roleplay, ça pourrait être une surexploitation d'une ressources (comme sur terre, si y'a plus de place pour mettre des champs et que la pop augmente, bah un moment la productivité/habitant diminue). Pareil pour les meca, simuler une absence de dispo à l'infini de ressources minières. J'ai fait en sorte que ça ne puisse pas être négatif (si le climat est favorable) donc pas de décroissance de la population par pénurie, mais un net income qui croit, passe par un max, puis redescent vers 0. Il faudrait donc savoir s'arrêter dans l'expansion de la pop sur une planète.

J'ai joué avec des équations pour voir la variation des paramètres actuellement envisagés et calculer un équilibrage, c'est présenté dans la feuille de calcul xls ici
https://github.com/LoucIsZbib/SpaceBaracraftCivilization/blob/main/game%20balance.ods 
WF = working force (pensez au chinois dans sa rizière)
RO = robots ou FA = Factory

Donnez moi votre avis si ça vous semble cohérent, pas trop compliqué sur le concept (les équations seront cachées au joueur, il verra que ce qu'il peut en tirer), intéressant, quelles pistes on pourrait envisager pour atteindre l'objectif initial : que les espèces aient des intérêts pour différentes planètes (et avoir des zergs !)

**Comment by odoucet** [comment]
Très bonne idée d'avoir plusieurs types de nations, mais pour moi je virerais les hybrides qui vont te compliquer fortement les algos (les 50/50 n'aurons que des planètes "moyennes"). 
Pour les courbes sans regarder dans le détail, c'est une bonne chose que la production diminue avec le temps (ou plutôt que l'effort nécessaire pour produire X% de plus augmente de plus en plus.

Si tu veux complexifier ton algo, tu peux aussi faire augmenter la température en fonction de la pop ^^

**Comment by LoucIsZbib** [comment]
oui, j'ai déjà pensé dans un futur d'inclure un réchauffement climatique :)

L'idée de staturer les planètes, c'est de limiter l'effet croissance exponentielle propre à ce type de jeu (et notamment FarHorizons), où je vois plusieurs travers : 
- il est parfois plus rentable de faire de la recherche MI MA que des expands
- les écarts entre les joueurs s'accentuent avec le temps, ils deviennent impossible à combler

Pour l'instant, l'aglo choisi est vraiment une cloche : au delà d'une pop optimale, le revenu net diminue.
On pourrait imaginer un truc qui continue à croitre mais à un prix de plus en plus cher. Un algo avec une croissance en racine carrée ? j'ai pas encore trouvé de truc idéal. Le code sera fait de sorte à pouvoir changer ça sans tout casser le reste du jeu

**Comment by odoucet** [comment]
je garderais une croissance en racine carré plutôt que du négatif, que les
joueurs auront du mal à comprendre. ça oblige les joueurs à s'étendre passé
une certaine taille.
Tu peux par contre prévoir un algo de resources "finies" par planète (et
secret), donc qu'au bout d'un certain temps on mine de moins en moins, non
pas parce qu'on a de la pop, mais parce que la planète est vide :)
pour l'équilibrage du jeu, toutes les planètes de départ devront avoir le
même nombre de ressources ; la répartition aléatoire ne devant toucher que
les autres planètes.

Le mer. 24 mars 2021 à 23:01, LoucIsZbib ***@***.***> a
écrit :
